app/src/main/java/com/demoapp/extension/ViewModelExtension.kt

```diff
@@ -0,0 +1,42 @@
+package com.demoapp.extension
+
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.*
+
+
+/**
+ * Generic fragment extension function to be able to retrieve the viewModel given a [T] class.
+ */
+inline fun <reified T : ViewModel> Fragment.getViewModel(factory: ViewModelProvider.Factory, body: T.() -> Unit = {}): T {
+    val vm = ViewModelProviders.of(this, factory)[T::class.java]
+    vm.body()
+    return vm
+}
+
+
+inline fun <reified T : ViewModel> AppCompatActivity.getViewModel(factory: ViewModelProvider.Factory, body: T.() -> Unit = {}): T {
+    val vm = ViewModelProviders.of(this, factory)[T::class.java]
+    vm.body()
+    return vm
+}
```

Comment:
We don't need this. it's already there in latest Kotlin version `viewModel()` and `activityViewModel()`

Reply:
removed ViewModelExtension.kt class